var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -254,6 +254,13 @@ def determine_b2_options(self, spec, options):
                 'toolset=%s' % self.determine_toolset(spec)
             ])
 
+        # clang is not officially supported for pre-compiled headers
+        # and at least in clang 3.9 still fails to build
+        #   http://www.boost.org/build/doc/html/bbv2/reference/precompiled_headers.html
+        #   https://svn.boost.org/trac/boost/ticket/12496
+        if spec.satisfies('%clang'):
```

Comment:
I see your point and have no strong opinion on adding or leaving it.
The main point is: leaving it disabled *until* it works will not be a dramatic drawback besides a bit of compile-time (if at all). Assuming it will be fixed in the next stable release that is not yet finished could be a bit vague. But let me spack-build a clang trunk!